configure_bazel.py

```diff
@@ -18,12 +18,51 @@
 import sys
 
 
+def detect_unix_platform_config(bazelrc):
+  # This is hoaky. Ideally, bazel had any kind of rational way of selecting
+  # options from within its environment (key word: "rational"), but sadly, it
+  # is unintelligible to mere mortals. Why should a build system have a way for
+  # people to condition their build options on what compiler they are using
+  # (without descending down the hole of deciphering what a Bazel toolchain is)?
+  # All I want to do is set a couple of project specific warning options!
+
+  if platform.system() == "Darwin":
+    print(f"build --config=generic_clang", file=bazelrc)
+    print(f"build:release --config=generic_clang_release", file=bazelrc)
+
+  # If the user specified a CXX environment var, bazel will later respect that,
+  # so we just see if it says "clang".
+  cxx = os.environ.get("CXX")
+  cc = os.environ.get("CC")
+  if (cxx is not None and cc is None) or (cxx is None and cc is not None):
+    print("WARNING: Only one of CXX or CC is set, which can confuse bazel. "
+          "Recommend: set both appropriately (or none)")
+  if cc is not None and cxx is not None:
+    # Persist the variables.
+    print(f"build --action_env CC=\"{cc}\"", file=bazelrc)
+    print(f"build --action_env CXX=\"{cxx}\"", file=bazelrc)
```

Comment:
I have a suspicion that this is going to make cache sharing with users invoking RBE not work at all

Reply:
I have no way to verify that. Feel free to take it back out if it is an issue.